core/src/main/java/de/fraunhofer/ids/messaging/core/config/ClientProvider.java

```diff
@@ -180,8 +180,17 @@ private static void setProxySelector(final OkHttpClient.Builder okHttpBuilder,
                     if (log.isDebugEnabled()) {
                         log.debug("Address: [host=({})], Port: [port=({})]", proxyHost, proxyPort);
                     }
-                    proxyList.add(new Proxy(Proxy.Type.HTTP,
-                        new InetSocketAddress(proxyHost, proxyPort)));
+
+                    if (proxyHost == null || proxyHost.trim().equals("")) {
+                        if (log.isWarnEnabled()) {
+                            log.warn("Proxy hostname invalid! Trying to skip using this proxy!"
```

Comment:
`proxyList.add(Proxy.NO_PROXY);` should be added here, or else the proxylist will be empty for this URI